Deploy-CAM.ps1

```diff
@@ -2709,6 +2661,155 @@ function Write-Host-Warning() {
+                }
+            } elseif ($radiusServerHost) {
+                $radiusConfig.radiusServerHost = $radiusServerHost
+            }
+
+            if ((-not $radiusServerPort) -and (-not $ignorePrompts)) {
+                if((confirmDialog "Do you want to change your RADIUS Server Port?") -eq 'y') {
+                    do {
+                        try {
+                            $radiusConfig.radiusServerPort = [int](Read-Host  "Enter your RADIUS Server's Listening port")
+                        } catch {
+                            $radiusConfig.radiusServerPort = $null
+                            Write-Host-Warning "Entered port is not an Integer"
+                        }                            
+                        if ( ($radiusConfig.radiusServerPort -le 0) -or ($radiusConfig.radiusServerPort -gt 65535) ) {
+                            Write-Host-Warning "Entered port is invalid. It should be between 1 and 65535."
+                            $radiusConfig.radiusServerPort = $null
+                        }            
+                    } while (-not $radiusConfig.radiusServerPort )
+                }
+            } elseif ( $radiusServerPort ) {
+                $radiusConfig.radiusServerPort = $radiusServerPort
+                $portValid = $false
+                do {
+                    if ( ($radiusConfig.radiusServerPort -le 0) -or ($radiusConfig.radiusServerPort -gt 65535) ) {
+                        Write-Host-Warning "Entered port is invalid. It should be between 1 and 65535."
+                        $portValid = $false
+                    } else {
+                        $portValid=$true
+                    }
+                    if (-not $portValid ) {
+                        try {
+                            $radiusConfig.radiusServerPort = [int](Read-Host  "Enter your RADIUS Server's Listening port")
+                        } catch {
+                            $portValid = $false
+                            Write-Host-Warning "Entered port is not an Integer"
+                        }
+                    }
+                } while (-not $portValid )
+            }
+
+            if ((-not $radiusSharedSecret) -and (-not $ignorePrompts)) {
+                if((confirmDialog "Do you want to change your RADIUS Server Shared Secret?") -eq 'y') {
+                    do {
+                        $radiusConfig.radiusSharedSecret = Read-Host -AsSecureString "Enter your RADIUS Server's Shared Secret"
+                    } while (-not $radiusConfig.radiusSharedSecret )
+                }
+            } elseif ( $radiusSharedSecret ) {
+                $radiusConfig.radiusSharedSecret = $radiusSharedSecret
+            }
+        }
+
+        # Store required keys as CamConfig Object
+        $camConfig = @{
+            parameters=@{}
+        }
+        foreach ($key in $radiusConfig.keys) {
+            if ($key -eq "radiusSharedSecret") {
```

Comment:
Can we just check type?

Reply:
I'm not sure what you mean

Reply:
I was just thinking - For each key if it's not a secure string make it a secure string. Don't check based on data structure key names. Then this code doesn't really care about the data structure.

Reply:
you're right, thats better